modules/python/src2/hdr_parser.py

```diff
@@ -958,7 +958,7 @@ def parse(self, hname, wmode=True):
                         else:
                             decls.append(decl)
 
-                            if self._generate_gpumat_decls and "cv.cuda" in decl[0]:
+                            if self._generate_gpumat_decls and any(x in decl[0] for x in ("cv.cuda", "imshow")):
```

Comment:
This condition looks weak.
Please add this `decl[0] == 'cv.imshow'` instead.

---

Or in this form for easy expansion (one new line per case):

```
delc[0] in [
    'cv.imshow',  # https://github.com/opencv/opencv/issues/18553
]
```